src/main/java/reposense/commits/CommitInfoAnalyzer.java

```diff
@@ -93,9 +101,65 @@ public static CommitResult analyzeCommit(CommitInfo commitInfo, RepoConfiguratio
             extractTagNames(tags);
         }
 
-        int insertion = getInsertion(statLine);
-        int deletion = getDeletion(statLine);
-        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, insertion, deletion);
+        if (statLine.isEmpty()) { // empty commit, no files changed
+            return new CommitResult(author, hash, date, messageTitle, messageBody, tags, 0, 0, new HashMap<>());
+        }
+
+        String[] statInfos = statLine.split(NEW_LINE_SPLITTER);
+        String[] fileTypeContributions = Arrays.copyOfRange(statInfos, 0, statInfos.length - 1);
+        Map<FileType, Map<String, Integer>> fileTypeAndContributionMap =
+                getFileTypesAndContribution(fileTypeContributions, config);
+
+        String contributionStat = statInfos[statInfos.length - 1]; // last index is the file contribution statistics
+        int totalCommitInsertions = getInsertion(contributionStat);
+        int totalCommitDeletions = getDeletion(contributionStat);
+
+        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, totalCommitInsertions,
+                totalCommitDeletions, fileTypeAndContributionMap);
+    }
+
+    /**
+     * Extract the additions and deletions of file types that have been modified
+     */
+    private static Map<FileType, Map<String, Integer>> getFileTypesAndContribution(String[] filePathContributions,
+            RepoConfiguration config) {
+        Map<FileType, Map<String, Integer>> fileTypesAndContribution = new HashMap<>();
+        for (String filePathContribution : filePathContributions) {
+
+            String[] infos = filePathContribution.split(TAB_SPLITTER);
+            int addition = Integer.parseInt(infos[0]);
+            int deletion = Integer.parseInt(infos[1]);
+            String filePath = extractFilePath(infos[2]);
```

Comment:
I'm personally fine with magic numbers here as the variable names are quite clear, could declare them as constants if it isn't intentional though

Reply:
Yep, declaring them as constants will be clearer